consensus/state.go

```diff
@@ -109,9 +115,10 @@ type State struct {
 	internalMsgQueue chan msgInfo
 	timeoutTicker    TimeoutTicker
 
-	// information about about added votes and block parts are written on this channel
+	// information about added votes and block parts, and bad peers, are written on this channel
 	// so statistics can be computed by reactor
-	statsMsgQueue chan msgInfo
+	// Takes a `msgInfo` or a `badPeerInfo`
+	peerInfoQueue chan interface{}
```

Comment:
This should be two distinct channels, not a single channel of `interface{}`.

Reply:
Go is not a dynamically typed language, `interface{}` is a huge red flag, not a pattern.

Reply:
Yeh I agonized over this a bit and wrote about the problem in the opening description. I thought introducing a channel would be more invasive but on second thought it should be pretty straight forward and would preserve better type discipline so you're probably right, thanks.